src/app/deliveries/page.jsx

```diff
@@ -62,16 +65,58 @@ export default function DeliveriesList() {
 	useEffect(() => {
 		const fetchData = async () => {
 			try {
-				const data = await fetchDeliveries(perPage, (page - 1) * perPage)
-				setData(data)
+				const data1 = await fetchDeliveries(perPage, (page - 1) * perPage)
+				setData(data1.elements)
+				let filteredDeliveries = data1.elements
+				if (startDate && endDate) {
+					filteredDeliveries = data.filter(delivery => {
+						const deliveryDate = new Date(delivery.date)
+						return (
+							deliveryDate >= new Date(startDate) &&
+							deliveryDate <= new Date(endDate)
+						)
+					})
+				} else if (startDate && !endDate) {
+					filteredDeliveries = data.filter(delivery => {
+						const deliveryDate = new Date(delivery.date)
+						return deliveryDate >= new Date(startDate)
+					})
+				} else if (!startDate && endDate) {
+					filteredDeliveries = data.filter(delivery => {
+						const deliveryDate = new Date(delivery.date)
+						return deliveryDate <= new Date(endDate)
+					})
+				}
+				setFilteredData(filteredDeliveries)
 			} catch (error) {
 				alert(
 					'Se produjo un error al cargar los datos. Por favor, inténtalo de nuevo.'
 				)
 			}
 		}
 		fetchData()
-	}, [page, perPage])
+	}, [page, perPage, startDate, endDate])
+
+	const handleSearch = searchTerm => {
+		console.log(searchTerm)
+		if (!searchTerm) {
+			setData(data)
+			setFilteredData(data)
+		} else {
+			if ('entregado'.includes(searchTerm.toLowerCase()))
+				searchTerm = 'delivered'
+			if ('avisado'.includes(searchTerm.toLowerCase())) searchTerm = 'notified'
+			if ('próximo'.includes(searchTerm.toLowerCase())) searchTerm = 'next'
+			const filtered = data.filter(
+				delivery =>
+					delivery.state.toLowerCase().includes(searchTerm.toLowerCase()) ||
+					names[delivery.family_id]
+						.toLowerCase()
+						.includes(searchTerm.toLowerCase())
+			)
+			setFilteredData(filtered)
```

Comment:
It would be nice if it also filtered by product name. For example if I type "arroz" it gives me deliveries with products containing that string